exams/utils.py

```diff
@@ -88,6 +89,29 @@ def _has_paid_for_exam(mmtrack, edx_course_key):
     return suppress_payment_for_exam or mmtrack.has_paid(edx_course_key)
 
 
+def _has_passed_course(mmtrack, edx_course_key):
```

Comment:
just to be precise, this is not a `classmethod` this is a function at the module level and I do not understand why we need it: how do you suppress payment?

if the purpose is just to wrap it in a try except, you can wrap the call to `extract_final_grade` inside `has_passed_course` and fix the tests around. I did not do it because I thought it was ok to raise, but it looks like it is better to do it.